quadbin/main.py

```diff
@@ -0,0 +1,534 @@
+    return (
+        0x4000000000000000
+        | (1 << 59)
+        | (z << 52)
+        | ((x | (y << 1)) >> 12)
+        | (0xFFFFFFFFFFFFF >> (z * 2))
+    )
+
+
+def cell_to_point(cell, geojson=False):
+    """Convert a cell into a geographic point.
+
+    Parameters
+    ----------
+    cell : int
+    geojson : bool, optional
+        Return the point as a GeoJSON geometry, by default False.
+
+    Returns
+    -------
+    str
+        if geojson is True.
+    list
+        if geojson is False (default).
+    """
+    tile = cell_to_tile(cell)
+
+    longitude = tile_to_longitude(tile, 0.5)
+    latitude = tile_to_latitude(tile, 0.5)
+
+    if geojson:
+        return '{{"type": "Point", "coordinates": {0}}}'.format([longitude, latitude])
+    else:
+        return [longitude, latitude]
+
+
+def point_to_cell(longitude, latitude, resolution):
+    """Convert a geographic point into a cell.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees.
+    latitude : float
+        Latitude in decimal degrees.
+    resolution : int
+        The resolution of the cell.
+
+    Returns
+    -------
+    int
+
+    Raises
+    ------
+    Exception
+        If the resolution is out of bounds.
+    """
+    if resolution < 0 or resolution > 26:
+        raise Exception("Invalid resolution: should be between 0 and 26")
+
+    longitude = clip_longitude(longitude)
+    latitude = clip_latitude(latitude)
+
+    tile = point_to_tile(longitude, latitude, resolution)
+
+    return tile_to_cell(tile)
+
+
+def cell_to_boundary(cell, geojson=False):
+    """Convert a cell into a geographic polygon.
+
+    Parameters
+    ----------
+    cell : int
+    geojson : bool, optional
+        Return the polygon as a GeoJSON geometry, by default False.
+
+    Returns
+    -------
+    str
+        if geojson is True.
+    list
+        if geojson is False (default).
+    """
+    bbox = cell_to_bounding_box(cell)
+
+    boundary = [
+        [bbox[0], bbox[3]],
+        [bbox[0], bbox[1]],
+        [bbox[2], bbox[1]],
+        [bbox[2], bbox[3]],
+        [bbox[0], bbox[3]],
+    ]
+
+    if geojson:
+        return '{{"type": "Polygon", "coordinates": [{0}]}}'.format(boundary)
+    else:
+        return boundary
+
+
+def cell_to_bounding_box(cell):
+    """Conver a cell into a geographic bounding box.
+
+    Parameters
+    ----------
+    cell : int
+
+    Returns
+    -------
+    list
+        Bounding box in degrees [xmin, ymin, xmax, ymax]
+    """
+    tile = cell_to_tile(cell)
+
+    xmin = tile_to_longitude(tile, 0)
+    xmax = tile_to_longitude(tile, 1)
+    ymin = tile_to_latitude(tile, 1)
+    ymax = tile_to_latitude(tile, 0)
+
+    return [xmin, ymin, xmax, ymax]
+
+
+def get_resolution(index):
+    """Get the resolution of an index.
+
+    Parameters
+    ----------
+    index : int
+
+    Returns
+    -------
+    int
+    """
+    return (index >> 52) & 0x1F
+
+
+def index_to_string(index):
```

Comment:
It returns a string but it's not really the string representation of the index. It's the hexadecimal representation. Why not call the function `index_to_hex` ? At least we need to specify this in the doc

Reply:
It's based on this one: https://h3geo.org/docs/api/inspection#h3tostring.

By definition, the string version of the index is the hex representation.

Reply:
Ok. What about a note in doc string specifying something like "The resulting string is the hexadecimal representation of the input integer".

Reply:
Yes, this is something "internal" of the spec, but we can add it in the docstring too :+1: